src/transformers/tokenization_utils_base.py

```diff
@@ -1550,8 +1551,29 @@ def __init__(self, **kwargs):
             {}
         )  # Use to store when we have already noticed a deprecation warning (avoid overlogging).
         self._in_target_context_manager = False
+        self.prompt_config = (
+            PromptConfig.from_dict(self.default_prompt_config, **kwargs) if self.can_generate else None
+        )
         super().__init__(**kwargs)
 
+    @property
+    def can_generate(self):
+        # TODO Figure out how to get this as the tokenizer
+        return True
```

Comment:
Not really necessary no? We can probably default the values in the prompt config, and only init if there is one uploaded. Each model can also initialise it to a default value wdyt?